src/aspire/abinitio/commonline_base.py

```
@@ -17,7 +17,14 @@ class CLOrient3D:
     """
 
     def __init__(
-        self, src, n_rad=None, n_theta=360, n_check=None, max_shift=0.15, shift_step=1
+        self,
+        src,
+        n_rad=None,
+        n_theta=360,
+        n_check=None,
+        max_shift=0.15,
+        shift_step=1,
+        mask=True,
```

I saw you mentioned this in the PR summary.... I too am debating if `mask` should be more flexible.  

Perhaps it should take an object.  Maybe a lightweight `Mask` class (mainly just to handle consistent grids and interface).  If the masks are in Fourier domain, can try to use existing `Filter`.  I don't think ASPIRE has anything similar for real domain.  The other place we mask are simple hard masks, like for noise estimation.

Anyway. False would be obviously be no mask, but I'm not sure how to express the default to be a mask.  Saying `mask=None` applies a mask feels sort of icky.

Ideas? Maybe sort of thing to discuss in dev meeting.  Really depends how important this masking is IMO.  If its just for completion and we don't plan to use much/extend in future, this PR is more than good enough.

---

Yeah, I wasn't sure. I don't know if really needs to be super flexible. It may just be that it's better to use the mask in some cases (ie. noisy images) and not in others (ie. clean images, class averages ?). So a simple on/off button might be enough.

Either way, I'm open to discussion on it. 

FYI, this mask is applied to the images in real space.

---

Okay, let's see if Joakim wants to bring it up.  Otherwise it is fine by me for now. Thanks!